django/core/serializers/python.py

```diff
@@ -96,100 +96,120 @@ def getvalue(self):
+                if self.ignorenonexistent and field_name not in field_names:
+                    # skip fields no longer on model
+                    continue
+
+                field = Model._meta.get_field(field_name)
+
+                # Handle M2M relations
+                if field.remote_field and isinstance(
+                    field.remote_field, models.ManyToManyRel
+                ):
+                    try:
+                        values = self._handle_m2m_field(field, field_value)
+                        if values == base.DEFER_FIELD:
+                            deferred_fields[field] = field_value
+                        else:
+                            m2m_data[field.name] = values
+                    except base.M2MDeserializationError as e:
+                        raise base.DeserializationError.WithData(
+                            e.original_exc, obj["model"], obj.get("pk"), e.pk
+                        )
+
+                # Handle FK fields
+                elif field.remote_field and isinstance(
+                    field.remote_field, models.ManyToOneRel
+                ):
+                    try:
+                        value = self._handle_foreign_key_field(field, field_value)
+                        if value == base.DEFER_FIELD:
+                            deferred_fields[field] = field_value
+                        else:
+                            data[field.attname] = value
+                    except Exception as e:
+                        raise base.DeserializationError.WithData(
+                            e, obj["model"], obj.get("pk"), field_value
+                        )
+
+                # Handle all other fields
+                else:
+                    try:
+                        data[field.name] = field.to_python(field_value)
+                    except Exception as e:
+                        raise base.DeserializationError.WithData(
+                            e, obj["model"], obj.get("pk"), field_value
+                        )
+
+            model_instance = base.build_instance(Model, data, self.using)
+            yield base.DeserializedObject(model_instance, m2m_data, deferred_fields)
+
+    def _handle_m2m_field(self, field, field_value):
+        return base.deserialize_m2m_values(
+            field, field_value, self.using, self.handle_forward_references
+        )
 
-def _get_model(model_identifier):
-    """Look up a model from an "app_label.model_name" string."""
-    try:
-        return apps.get_model(model_identifier)
-    except (LookupError, TypeError):
-        raise base.DeserializationError(
-            "Invalid model identifier: '%s'" % model_identifier
+    def _handle_foreign_key_field(self, field, field_value):
+        return base.deserialize_fk_value(
+            field, field_value, self.using, self.handle_forward_references
         )
+
+    @staticmethod
+    def _get_model(model_identifier):
```

Comment:
If we look at `xml.Deserializer` this has `_get_model_from_node`, `_handle_m2m_field_node`, `_handle_fk_field_node` and `__next__` instead of `__iter__` is defined - it would be good to align this